aws/sns/resources.py

```diff
@@ -21,20 +22,49 @@ def sns_topics():
     )
 
 
-def sns_subscription_attributes():
+def sns_parent_topic_exists(subscription):
     "https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#subscription"
-    return [
+    topicArn = ":".join(
+        re.split(":", subscription)[0:6]
+    )  # Naive method of divining topic from subscription
+    try:
         botocore_client.get(
             service_name="sns",
-            method_name="get_subscription_attributes",
+            method_name="get_topic_attributes",
             call_args=[],
-            call_kwargs={"SubscriptionArn": subscription["SubscriptionArn"]},
-            profiles=[subscription["__pytest_meta"]["profile"]],
-            regions=[subscription["__pytest_meta"]["region"]],
+            call_kwargs={"TopicArn": topicArn},
+        )
+        return True
+    except:
```

Comment:
It could be helpful to be a bit more specific what errors to catch – catching, e.g., `TypeError` here could make for some fun debugging.

Reply:
I'm going to take a stab at checking the topics list instead so catching this error should no longer be required.  If we have to go this route I'll add TypeError here.